processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1010,6 +1024,17 @@ public QueryableIndex load(File inDir) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
+      Map<String, Object> metadata = null;
+      ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
+      if (metadataBB != null) {
+        metadata = mapper.readValue(
+            serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
```

Comment:
That also seems to be how it's handled elsewhere: null metadata is OK until someone really needs it.